README.md

```diff
@@ -1,46 +1,123 @@
 # Rails URL Helpers
 
-## Objectives
+Since Rails by design was meant to be flexible, the result is that there are typically a number of ways to accomplish the same features. Routes are a great example for how this principle operates in a Rails app. As you have already seen in our Capybara feature specs, we have the ability to hardcode paths or to use view helper methods. In this section we will extend this knowledge and walk though both options and how they can be used in the implementation code itself.
 
-1. Describe why route helpers are useful (relying on abstractions such as methods as opposed to literring our application with URL data)
-2. Use rake routes to find the name of a route
-3. Predict route names based on rails routing conventions
-4. Use link_to to generate an <a>
-5. Use _path and _url helpers to generate URLs
-6. Name routes with :as option
-7. Generate URLs with route helpers that require variables.
 
+## Paths vs Route Helpers
 
-## Notes
+So why would we want to use route helper methods as opposed to hard coding paths into the application? There are a number of reasons, below are a few of the key rationales:
 
-Given a posts#index and a posts#show, they should link between each other
+* Route helper methods help clean up the view and controller code and assist with readability. On a side note, you cannot use these helper methods in your model files.
 
-we could use `<a href="/posts/#{@post.id}">` but thats very low level.
+* It's more natural to be able to pass arguments into a method as opposed to using string interpolation. For example, using ```post_path(post, opt_in: true)``` is more readable than using ```"posts/<%= post.id %>?opt_in=true"```
 
-Let's start by having posts#index link to individual posts show pages.
+* Route helpers are more dynamic since they are methods and not simply strings
 
-link_to instead of a href but lets still use interpolation.
+* Route helpers translate directly into HTML friendly paths
 
-Then let's introduce the idea of named routes and instead of hardcoding URLs we'll let our URL Helpers generate them.
 
-what are the _path / _url helpers and when / where in teh stack can you use them (views/controllers yes, models no)
+## Implementing Route Helpers
 
-Why? Maintainability, relying on abstractions.
+To begin, we're going to start with an application that has the MVC setup for ```posts```, with ```index``` and ```show``` actions currently in place. The route call looks like this:
 
-rake routes
+```ruby
+# config/routes.rb
+resources :posts, only: [:index, :show]
+```
 
-how routes get names, the default
+We briefly discussed this ```resources``` method in the dynamic routing lesson, this will create routing methods for posts that we can utilized in our views and controllers. By running ```rake routes``` in the terminal it will give the following output:
```

Comment:
I would include a rails app with these resources already set up so students can run that code in their terminal and follow along